### ISSUE TITLE
Constant definition

### ISSUE THREAD

**Issue by DolajoCZ** [microsoft/pyright#1988]
Hello, I've started using type check in VScode in strict mode. Unfortunatelly I get following error. Is this valid error? For each case is this constant defined and for example mypy is ok with such definition. Thanks

![image](https://user-images.githubusercontent.com/23177301/122055240-b6540b00-cde8-11eb-8fcc-a25817444b23.png)

**Comment by erictraut** [comment]
This is as intended. The `reportConstantRedefinition` verifies that a constant has only one declaration.

I don't think mypy supports a similar rule, which is why you are not seeing an error there.

You have several options here:

1. If you don't like the way this works, you can disable the `reportConstantRedefinition` rule globally or for a single file using a `# pyright: reportConstantRedefinition=false` comment.
2. You can suppress the error using a `# type: ignore` comment.
3. You can change the name of the variable so it is not a constant.
4. You can use a form of the `sys.platform` check that is understood statically by pyright (and other type checkers) — namely a simple `==` check, like this:

```python
if sys.platform == "win32":
    CURRENT_OS = "windows"
elif sys.platform == "linux":
    CURRENT_OS = "linux"
elif sys.platform == "darwin":
    CURRENT_OS = "mac"
else:
    raise Exception("Unknown operating system")
```

**Comment by DolajoCZ** [comment]
Thank you @erictraut for quick response, explanation and suggestions how to solve this issue.
